### ISSUE TITLE
The linter blows up processing zips using '\' as path separators.

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-linter#1426]
Using https://github.com/mozilla/addons/files/1173219/InteractSyncForFirefox.zip

Running the validator directly locally outputs:

```
$ bin/addons-linter ~/Downloads/InteractSyncForFirefox.zip
Error: invalid characters in fileName: common\AnimatorFactory.js
    at /Users/whatever/code/addons-linter/node_modules/yauzl/index.js:413:70
    at /Users/whatever/code/addons-linter/node_modules/yauzl/index.js:618:5
    at /Users/whatever/code/addons-linter/node_modules/fd-slicer/index.js:32:7
    at FSReqWrap.wrapper [as oncomplete] (fs.js:682:17)
```

This exception should be caught and exposed to the user. Currently when validating on AMO we show:

<img width="1033" alt="upload_add-on____developer_hub____add-ons_for_firefox" src="https://user-images.githubusercontent.com/1514/28622233-b9f178a2-720b-11e7-8e2f-4bc7f17e255c.png">

The [zip spec says](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT):

>   4.4.17 file name: (Variable)
> 
>        4.4.17.1 The name of the file, with optional relative path.
>        The path stored MUST not contain a drive or
>        device letter, or a leading slash.  All slashes
>        MUST be forward slashes '/' as opposed to
>        backwards slashes '\' for compatibility with Amiga
>        and UNIX file systems etc.  If input came from standard
>        input, there is no file name field.  

However we've seen a couple of cases where archives created on windows use `\` as a separator:

- https://github.com/mozilla/addons/issues/450
- https://github.com/mozilla/addons/issues/442

Microsoft have provided a fix https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/mitigation-ziparchiveentry-fullname-path-separator but we should look to provide a better warning anyway.

**Comment by muffinresearch** [comment]
We could possibly need a bug filed on the server too if that opens the zip before processing the validation.

**Comment by EnTeQuAk** [comment]
> We could possibly need a bug filed on the server too if that opens the zip before processing the validation.

I added https://github.com/mozilla/addons-server/issues/6111

**Comment by muffinresearch** [comment]
Wontfix
